ksql-engine/src/main/java/io/confluent/ksql/materialization/Row.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.materialization;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.errorprone.annotations.Immutable;
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.materialization.TableRowValidation.Validator;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.kafka.connect.data.Struct;
+
+@Immutable
+public final class Row implements TableRow {
+
+  private final LogicalSchema schema;
```

Comment:
Seeing that this is an immutable struct like class, why not just make the fields public and get rid of the accessors?